src/service/position-observer-impl.js

```diff
@@ -92,13 +124,32 @@ class AbstractPositionObserver {
           Math.floor(Math.random() * LOW_FIDELITY_FRAME_COUNT) : 0,
       trigger(position) {
         const prePos = entry.position;
+
         if (prePos
             && layoutRectEquals(prePos.positionRect, position.positionRect)
             && layoutRectEquals(prePos.viewportRect, position.viewportRect)) {
           // position doesn't change, do nothing.
           return;
         }
+
+        // Add the relative position regardless of whether the element
+        // is inside the view or not
+        position.relativePos = layoutRectsRelativePos(
```

Comment:
position should be a `PositionInViewportEntryDef` object. If we are assigning extra param to position, we will need to update the definition of `PositionInViewportEntryDef` object as well